core/includes/database/database.inc

```diff
@@ -2471,20 +2625,20 @@ function db_query($query, array $args = array(), array $options = array()) {
 /**
  * Executes a query against the active database, restricted to a range.
  *
- * @param $query
+ * @param string $query
  *   The prepared statement query to run. Although it will accept both named and
  *   unnamed placeholders, named placeholders are strongly preferred as they are
  *   more self-documenting.
- * @param $from
+ * @param int $from
  *   The first record from the result set to return.
- * @param $count
+ * @param int $count
  *   The number of records to return from the result set.
- * @param $args
+ * @param array $args
  *   An array of values to substitute into the query. If the query uses named
  *   placeholders, this is an associative array in any order. If the query uses
  *   unnamed placeholders (?), this is an indexed array and the order must match
  *   the order of placeholders in the query string.
- * @param $options
+ * @param array $options
  *   An array of options to control how the query operates.
  *
  * @return DatabaseStatementInterface|DatabaseStatementBase|DatabaseStatementPrefetch|stdClass[]
```

Comment:
These return types seem a bit odd. But maybe I'm misunderstanding things...

Abstract public function queryRange() claims its returns:
```
   * @return DatabaseStatementInterface|DatabaseStatementBase|DatabaseStatementPrefetch
   *   A database query result resource, or NULL if the query was not executed
   *   correctly.
```
^^ and null is only mentioned in comment text.

No idea, where the stdClass array comes from. This doesn't necessarily mean I'm right, I only scratch my head... :wink:

Reply:
No idea either. Doesn't seem right.

Reply:
I aggressively added type hints back in https://github.com/backdrop/backdrop-issues/issues/1937. But introspection tools have gotten a lot better now and I don't think the `stdClass[]` thing applies. Or maybe that was added just because it silenced warnings about undefined properties being accessed. I don't think it is needed.